public/app/features/correlations/utils.ts

```diff
@@ -31,7 +31,9 @@ export const attachCorrelationsToDataFrames = (
       dataSourceUid = dataFrameRefIdToDataSourceUid[formattedRefID];
     }
 
-    const sourceCorrelations = correlations.filter((correlation) => correlation.source.uid === dataSourceUid);
+    const sourceCorrelations = correlations.filter((correlation) => {
+      return correlation.source ? correlation.source.uid === dataSourceUid : false;
+    });
```

Comment:
should we change the `CorrelationData` type accordingly?

Reply:
@Elfo404 actually, we can be smarter about this and filter out correlations that don't have sources when we fetch them. I'll do that instead.